source/book/include/marlin/book/Entry.h

```diff
@@ -0,0 +1,183 @@
+      friend BookStore ;
+
+      /// constructor
+      EntryMultiCopy( const Context &context ) : _context{context} {}
+
+    public:
+      /// default constructor
+      EntryMultiCopy() = default ;
+
+      /**
+       *	@brief creates a new handle for the object.
+       *	@param idx id of instance to get handle.
+       *	@attention Use one instance only in one thread at the same time.
+       */
+      Handle< T > handle( std::size_t idx ) {
+        return Handle( _context.mem, _context.mem->at< T >( idx ) ) ;
+      }
+
+    private:
+      /// \see {EntrySingle::_context}
+      Context _context ;
+    } ;
+
+    /**
+     *	@brief entry for object to be used Multithreaded.
+     *	contain only one Instance and a thread save way to write.
+     *	@note keep synchronisation points in mind.
+     */
+    template < typename T >
+    class EntryMultiShared : public EntryBase {
+      friend BookStore ;
+
+      /// constructor
+      EntryMultiShared( const Context &context ) : _context{context} {}
+
+    public:
+      /// default constructor
+      EntryMultiShared() = default ;
+
+      /**
+       *	@brief creates a new handle for the object.
+       *	Each handle may contain some kind of Buffer to reduce sync-points.
+       */
+      Handle< T > handle() {
+        return Handle( _context.mem, _context.mem->at< T >( 0 ) ) ;
+      }
+
+    private:
+      /// \see {EntrySingle::_context}
+      Context _context ;
+    } ;
+
+    /**
+     *	@brief class to store and manage objects in BookStore.
+     */
+    class Entry {
+      friend BookStore ;
+
+      /// constructor
+      Entry( const std::shared_ptr< EntryBase > &entry, const EntryKey &key )
+        : _key{key}, _entry{entry} {}
+
+      /// reduce Entry to default constructed version.
+      void clear() {
+        _key = EntryKey{} ;
+        _entry.reset() ;
+      }
+
+    public:
+      /// default constructor. Not valid!
+      Entry() = default ;
+
+      /**
+       *	@brief creates an handle for the entry.
+       *	@param idx of instance, only used for multi copy entries
+       *	@return empty optional if the type or the configuration not matches.
+       */
+      template < class T >
+      std::optional< Handle< T > > handle( std::size_t idx = -1 ) const {
+        if ( std::type_index( typeid( T ) ) != _key.type ) {
+          // TODO add log
+          return std::optional< Handle< T > >() ;
+        }
+
+        if ( _key.flags.Contains( Flags::Book::Single ) ) {
+          return std::static_pointer_cast< EntrySingle< T > >( _entry )
+            ->handle() ;
+
+        } else if ( _key.flags.Contains( Flags::Book::MultiCopy ) ) {
+          return std::static_pointer_cast< EntryMultiCopy< T > >( _entry )
+            ->handle( idx ) ;
+
+        } else if ( _key.flags.Contains( Flags::Book::MultiShared ) ) {
+          return std::static_pointer_cast< EntryMultiShared< T > >( _entry )
+            ->handle() ;
+        }
+        // TODO add log
```

Comment:
also, question, 
Should we log it if the creation failed because of a type mismatch or invalid Flag?
because when it's failed it's an internal error